src/compiler/_normalize-3d-spec.ts

```diff
@@ -18,13 +18,23 @@ export function _fixTrackToWalkaround(t: Track) {
             chr,
             coord
         };
-        const dataTransform: JoinTransform = {
+        const dataTransform: Omit<JoinTransform, 'to'> = {
             type: 'join',
-            from: { url, chromosomeField: chr, genomicField: coord },
-            to: { startField: 'start', endField: 'end' }
+            from: { url, chromosomeField: chr, genomicField: coord }
         };
+        // TODO: need to support other data types as well
+        let to: JoinTransform['to'];
+        if (t.data?.type === 'bigwig') {
+            to = { startField: 'start', endField: 'end' };
+        } else if (t.data?.type === 'csv') {
+            // TODO: Not the reliable way to select two data fields corresponding to start and end positions
+            const startField = t.data.genomicFields?.[0] ?? 'unknown';
+            const endField = t.data.genomicFields?.[1] ?? undefined;
+            to = { startField, endField };
+        }
```

Comment:
Users need to specify what are the genomic fields in the csv file (e.g., `genomicFields`: ['position']). With this change, we are using the first two fields in the join operation.

If only one genomic field is specified in the csv data (e.g., `genomicFields: ['position']`, the `join` operation compares the values of `position` and `coord` to connect the 3D genome model and the CSV data.

If two genomic fields are specified (e.g., `genomicFields: ['start', 'end']`, the `join` operation checks whether `coord` is within the range of 'start' and 'end'.